### ISSUE TITLE
.slice(..0) panics!

### ISSUE THREAD

**Issue by rendarz** [staticintlucas/stringslice#1]
If we have this very little Rust snippet:
```rust
fn main() {
    let s = String::from("abcd");
    println!("range `{}`", &s.as_str()[..0]);
}
```
We can see that if we pass the range `..0` to a `&str` slice, it works by returning an empty string slice.

If we do the same operation with your crate, e.g. `s.slice(..0)` it panics with an overflow.
Is this a bug? Shouldn't that follow the "common rules" of Rust slices ranges and returning an empty slice too?

Thank you.

UPDATE: This panic overflow happens also specifying range `0..0` which still works fine in normal &str slice.
UPDATE2: This panic overflow happens also specifying range `LEN..LEN` which - again - still works fine in normal &str slice.

**Comment by staticintlucas** [comment]
Yes, looks like this is a bug when taking zero-length slices. This originates in `try_substring` which returns `None` instead of `Some("")`.

I have fixed this in 39f74e3a7fb4e388f0cc9736ae597298608638c2 and I'll get a new release out later today. Since there are changes to the behaviour of public APIs, this will be v0.2.

**Comment by staticintlucas** [comment]
0.2 is now available on [crates.io](https://crates.io/crates/stringslice). Thanks for reporting!
